config/config.go

```diff
@@ -144,7 +146,7 @@ func ReadConfigFile(preCfg *Config) (*Config, error) {
 	// done. This prevents the warning on help messages and invalid
 	// options.
 	if configFileError != nil {
-		log.Warning("%v", configFileError)
+		logging.FromContext(context.Background()).Warn("failed to read config file", zap.Error(configFileError))
```

Comment:
does background context has any value here for debugging?

Reply:
No, `logging.FromContext(context.Background())` creates a new default logger (as there isn't any logger attached to the background context).

I refactored all code where it occurred to bubble up the error and handle it there.